fastfood/food.py

```diff
@@ -141,85 +150,29 @@ def update_cookbook(cookbook, templatepack, stencilset_name, **options):
-            if not isinstance(data, list):
-                raise TypeError("Base directories should be a list of "
-                                "directory names.")
-            directories = data
+    if not os.path.exists(cookbooks_home):
+        raise ValueError("Target cookbook dir %s does not exist."
+                         % os.path.relpath(cookbooks_home))
+
+    target_dir = os.path.join(cookbooks_home, cookbook_name)
+    LOG.debug("Creating dir -> %s", target_dir)
+    try:
+        os.makedirs(target_dir)
+    except OSError as err:
+        if err.errno != errno.EEXIST:
+            raise
         else:
-            raise ValueError("Unknown 'base' option '%s'" % option)
+            LOG.info("Skipping existing directory %s", target_dir)
 
-    if files:
-        if not os.path.exists(cookbooks_home):
-            raise ValueError("Target cookbook dir %s does not exist."
-                             % os.path.relpath(cookbooks_home))
-
-    template_map = {
-        'cookbook': {
-            'name': cookbook_name,
-            'year': datetime.datetime.now().year,
-            }
-        }
-
-    template_files = [os.path.join(templatepack, 'base', f)
-                      for f in files]
-    path_map = dict(zip(template_files, files))
-    filetable = templating.render_templates(*template_files, **template_map)
-
-    for path in directories:
-        target_dir = os.path.join(cookbooks_home, cookbook_name, path)
-        LOG.debug("Creating dir -> %s", target_dir)
-        try:
-            os.makedirs(target_dir)
-        except OSError as err:
-            if err.errno != errno.EEXIST:
-                raise
-            else:
-                LOG.info("Skipping existing directory %s",
-                         target_dir)
-
-    written_files = []
-    for orig_path, content in filetable:
-        target_path = os.path.join(
-            cookbooks_home, cookbook_name, path_map[orig_path])
-        needdir = os.path.dirname(target_path)
-        assert needdir, "Target should have valid parent dir"
-        try:
-            os.makedirs(needdir)
-        except OSError as err:
-            if err.errno != errno.EEXIST:
-                raise
-
-        if os.path.isfile(target_path):
-            if force:
-                LOG.warning("Forcing overwrite of existing file %s.",
-                            target_path)
-            else:
-                LOG.info("Skipping existing file %s", target_path)
-                continue
-        with open(target_path, 'w') as newfile:
-            LOG.info("Writing rendered file %s", target_path)
-            written_files.append(target_path)
-            newfile.write(content)
+    cookbook_path = os.path.join(cookbooks_home, cookbook_name)
+    cookbook = book.CookBook(cookbook_path)
 
-    return (written_files, book.CookBook(os.path.join(cookbooks_home,
-                                                      cookbook_name)))
+    return (cookbook, cookbook_path)
```

Comment:
Now that this doesn't return a list of written files, now you can just return the `Cookbook` instance and if the caller wants the path they can use `cookbook.path`.